### ISSUE TITLE
Petitioner: Select party type and enter appropriate contact information

### ISSUE THREAD

**Issue by mariahkannenberg** [flexion/ef-cms#307]
As a Petitioner, in order to inform the court about what type of party I'm filing as, I need to indicate my party type and provide the contact information that's required for that party type

The person filing a petition may be filing for themselves, them and a spouse, on behalf of someone (child, elderly parent (incompetent person), business, estate, trust, etc.). The Court needs the ability for the person who is filing the Petition to indicate who they are filing for and then provide contact information fields based on that response.

## Pre-Conditions
* Petitioner has confirmed that they want to Start a New Case

## Acceptance Criteria
* Petitioner must select a Party Type
* Party types must be official, court-designated party types [Provided in Notes]
* Petitioners are required to enter the contact information that's required for the specific party type they've selected
* Party Type and contact information is saved in the System
* Petitioner and internal court users can easily see within the case all party information

## Notes
* The Party Types are:
    * Petitioner
    * Petitioner & Spouse
    * Corporation
    * Partnership (as the Tax Matters Partner)
    * Partnership (as a Partner Other Than Tax Matters Partner)
    * Partnership (BBA Regime)
    * Estate with an Executor/Personal Representative/Fiduciary/etc.
    * Estate without an Executor/Personal Representative/Fiduciary/etc.
    * Trust
    * Conservator
    * Guardian
    * Custodian
    * Minor (Without a Guardian, Conservator, or other like Fiduciary)
    * Incompetent Person (Without a Guardian, Conservator, or other like Fiduciary)
    * Surviving Spouse
    * Petitioner and Surviving Spouse
    * Transferee
    * Donor

* Required contact information fields for each Party Type (https://docs.google.com/document/d/1_pv9J_RnUGVdEM8HOdX4iikx3rKbgQXyXd_OG8BVC_I/edit?usp=sharing)
* This story does not cover international addresses.  That will be covered in another story (#655). If the design team feels it should be included in the design then combine #655 .
* Jessica needs to learn more about how the Court treats aliases and how we need to capture that information in the system (if at all for MVP).

## Tasks
- [x] Design Party Type selection section of Start a Case Form (Kristen) 
- [x] Update design of Party Information on Case Information tab (Kristen) 
- [x] Develop business tests
- [x] Update design and case information tab
- [x] Design views based upon system flows
- [x] Modify party type entity and additional entity combinations
- [x] Modify Start a Case view
- [x] Field-level validation for existing fields (eric)
- [x] Field-level validation for petitioner contact (eric)
- [x] petitioner & spouse & deceased spouse validation (William)
- [x] estate with & without executor, validation (kris)
- [x] place input forms for each party type into StartCase (rachael)
- [x] corporation, partnership tax matters, partnership NOT tax matters entity validation (William)
- [x] partnership BBA regime, trust & trustee entity validation (kris)
- [x] conservator, guardian, custodian entity validation (kris)
- [x] minor without guardian, incompetent person without guardian entity validation (William)
- [x] surviving spouse, petitioner & deceased spouse entity validation (William)
- [x] donor, transferee entity validation (kris)
- [x] remove "email" field from all secondary users (Rachael)
- [x] verify/clean up mobile view (Rachael)
- [x] change case title based on party type (Rachael)
- [x] field-level validation on address (Eric)
- [x] Jessica's comments from review: (1) When you select “Myself & Spouse” and “Deceased=NO”, fill everything out and try to submit the case, there is an error that comes up requiring an email.  There is no blank email field, so this is a false-positive error. (2) Another false-positive error when you select “Other”…”with Estate/Trust”…”Estate with Executor/PR”.  This error message says Name is required (x2) and Phone is required. (3) Another false-positive error when you select “Other”…”Estate/Trust”…”NO Executor/PR”.   This error message says Name is required. (Rachael) - PR ready

## Definition of Done
- [x] acceptance criteria have been met
- [x] Field level and page level validation errors (front-end and server side) integrated and functioning
- [x] all code changes have been reviewed, approved, and merged and deployed to develop 
- [x] all automated unit tests are passing on all supported browsers with coverage > 90% 
- [x] run the browser stack scripts to verify that application is usable in Chrome, Firefox, Safari, Edge and Internet Explorer  
- [x] all automated a11y checks are passing - no new WCAG2.1 or AA contrast issues present
- [x] 0 errors reported in manual a11y testing
- [x] all new functionality verified to work with keyboard and osx voiceover
- [x] all code quality checks are passing - sonar, eslint, html validation, etc.
- [x] code refactored for clarity and to remove any known technical debt 
- [x] application can be deployed to multiple regions
- [x]  README, other appropriate docs, and swagger/APIs fully updated - all code functions documented with jsdocs
- [x] performance tests successfully completed and no new significant performance issues found
- [x] alerting and monitoring for new functionality in place and successfully exercised
- [x] ui should be touch optimized and responsive (functions on supported mobile devices and optimized for screen sizes as required)
- [x] usability has been validated
- [x] npm audit for CVE’s in production library/module dependencies identified and resolved
- [x] library/module dependencies are up-to-date and are at the latest resolvable version

**Comment by uxforgood** [comment]
Petitioner
Petitioner & Spouse

A Company
* Corporation
* Partnership (as the Tax Matters Partner)
* Partnership (as a Partner Other Than Tax Matters Partner)
* Partnership (as a Partnership Representative)

Estate or Trust
* Estate with an Executor/Personal Representative/Fiduciary/etc.
* Estate without an Executor/Personal Representative/Fiduciary/etc.
* Trust  
* Conservator  [[representatives authorized for these proceedings]]
* Custodian  [[representatives authorized for these proceedings]]

Other [[what is your relationship to Petitioner?]]
* Guardian
* Minor (Without a Guardian, Conservator, or other like Fiduciary)
* Surviving Spouse
* Incompetent Person (Without a Guardian, Conservator, or other like Fiduciary)

* Transferee  [[could be with a company - "sold company to avoid taxes" this is the people that sold it, not the ones who bought it]]  [[individual sells property and claims erroneous tax benefits]] [[gift taxes]]
* Donor [[charitable/nonprofit situation]]

**Comment by klohman** [comment]
@JessicaMarine 

Language for Party Type Selection 

### I am filing on behalf of ....
Myself, Test Taxpayer 
Myself ,Test Taxpayer, and my spouse 
A Business 
An Estate or Trust 
Other 


### What type of business are you representing? 
Corporation
Partnership (as the Tax Matters Partner)
Partnership (as a Partner Other Than Tax Matters Partner)
Partnership (BBA Regime)

### I am representing this estate or trust as a...
[Authorized Representative for an] Estate with an Executor/Personal Representative/Fiduciary/etc.
[Authorized Representative for an] Estate without an Executor/Personal Representative/Fiduciary/etc.
[Authorized Representative for a ]Trust
Conservator
Custodian 

### I am representing this taxpayer as a ...
Guardian
[Authorized Representative of a] Minor (Without a Guardian, Conservator, or other like Fiduciary)
[Authorized Representative of a] Incompetent Person (Without a Guardian, Conservator, or other like Fiduciary)
Surviving Spouse
Transferee
Donor

**Comment by klohman** [comment]
# Party Type: Petitioner 

## Default Option 

User will see the following options by default: 
- Myself 
- Myself and my spouse 
- A business 
- Other 

![307- Start a Case- Default.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/e63e3716-71c8-48c6-9ff2-23c2dc1c168b)

## Petitioner Party Type 

On select of "Myself" 
- the "party type" should be set to "Petitioner" 
- the following view "Tell Us About Yourself" should display 

![307- Start a Case- Myself.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/08a9e5bb-76b1-4bcc-b07b-26a91dde66b3)

### Contact Field Validation 
- All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4 
   - Numeric values only 
- Email Validation 
   - email will be pre-populated with the registered user's email 
   - email will be read-only

**Comment by klohman** [comment]
# Party Type: Petitioner and Spouse 

On select of "Myself and my spouse" 
   - display radio button field " Is your spouse deceased?" 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/ac0050c0-4f97-4897-a971-d70efdd7fe70)

On select of "No"
- party type is set to "Petitioner and Spouse" 
- "Tell Us About Yourself" and "Tell Us About Your Spouse" contact field views display 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/991403a6-9919-4167-aad3-e728176e94d5)

## Tell Us About Yourself (Petitioner) 
- Is Primary Contact 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only


## Tell Us About Your Spouse (Spouse) 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 
 
- Email validation 
  - must be valid email address format (prefix@domain)
  - error message for invalid email " Please enter a valid email address."

**Comment by klohman** [comment]
# Party Type: Petitioner and Deceased Spouse 

On select of "Myself and my spouse" 
   - display radio button field " Is your spouse deceased?" 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/ac0050c0-4f97-4897-a971-d70efdd7fe70)

On select of "Yes"
- party type is set to "Petitioner and Deceased Spouse" 
- "Tell Us About Yourself" and "Tell Us About Your Deceased Spouse" contact field views display 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/3004945b-a6ec-4484-af09-75dcd4bc36a9)

## Tell Us About Yourself (Primary Contact)

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only


## Tell Us About Your Deceased Spouse 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code."

**Comment by klohman** [comment]
# Party Type: Surviving Spouse  

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/42fc204e-7b1b-4a82-b4e1-bb7d5fb05aae)

On select of "Deceased Spouse"
- party type is set to "Surviving Spouse" 
- "Tell Us About Yourself as the Surviving Spouse" and "Tell Us About Your Decesaed Spouse" contact field views display 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/991403a6-9919-4167-aad3-e728176e94d5)

## Tell Us About Yourself as the Surviving Spouse (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only


## Tell Us About Your Deceased Spouse 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code."

**Comment by klohman** [comment]
# Party Type: Estate without Executor/Personal Representative/Etc. 

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "Estate or Trust", display the radio button options:
- Estate with an Executor/Personal Representative/Fiduciary/etc. 
- Estate without an Executor/Personal Representative/Fiduciary/etc. 
- Trust 

!![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/9e1c433d-1edb-41a4-a1a1-ea1455b2a596)


On select of "Estate without Executor/Personal Representative/Etc." radio button option 
- party type is set to "Estate without Executor/Personal Representative/Etc." 
- "Tell Us About the Estate You Are Filing For" view displays 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/d25004a4-78a1-42bf-9142-a1317f35aa32)

### Tell Us About the Estate You Are Filing For (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

**Comment by klohman** [comment]
# Party Type: Estate with Executor/Personal Representative/Etc. 

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "Estate or Trust", display the radio button options:
- Estate with an Executor/Personal Representative/Fiduciary/etc. 
- Estate without an Executor/Personal Representative/Fiduciary/etc. 
- Trust 

!![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/9e1c433d-1edb-41a4-a1a1-ea1455b2a596)


On select of "Estate with Executor/Personal Representative/Etc." radio button option 
- party type is set to "Estate with Executor/Personal Representative/Etc." 
- "Tell Us About Yourself as the Executor/Personal Representative For This Estate" and " Tell Us About the Estate You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/e016f59a-d4b5-42b4-b5e5-0d71c7adcfb2)

### Tell Us About Yourself as the Executor/Personal Representative For this Estate (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

## Tell Us About the Estate You Are Filing For (Secondary Contact) 
### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code."

**Comment by klohman** [comment]
# Party Type: Trust 

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "Estate or Trust", display the radio button options:
- Estate with an Executor/Personal Representative/Fiduciary/etc. 
- Estate without an Executor/Personal Representative/Fiduciary/etc. 
- Trust 

!![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/9e1c433d-1edb-41a4-a1a1-ea1455b2a596)


On select of "Trust" radio button option 
- party type is set to "Trust and Trustee" 
- "Tell Us About Yourself as the Trustee" and " Tell Us About the Trust You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/d633bec6-92e9-46f5-b718-1fe82b4a1a01)

### Tell Us About Yourself as the Trustee (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

## Tell Us About the Trust You Are Filing For (Secondary Contact) 
### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 
- Email validation 
   - email must be valid email format (prefix@domain) 
   - Error message for invalid email " Please enter a valid email address."

**Comment by klohman** [comment]
# Party Type: Conservator   

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "A minor or incompetent person", display the radio button options:
- Conservator 
- Guardian 
- Custodian 
- Next Friend for a Minor (Without a Guardian, Conservator, or other like Fiduciary)
- Next Friend for an Incompetent Person (Without a Guardian, Conservator, or other like Fiduciary)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b8c178dc-4730-40fe-913c-ac190abd0a9c)

On select of "Conservator" radio button option 
- party type is set to "Conservator" 
- "Tell Us About Yourself as the Conservator for This Taxpayer" and " Tell Us About the Taxpayer You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/bd59a260-fae6-4c21-bd51-215b5dd7e389)

### Tell Us About Yourself as the Conservator for This Taxpayer (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

## Tell Us About the Taxpayer You Are Filing For (Secondary Contact) 
### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 
- Email validation 
   - email must be valid email format (prefix@domain) 
   - Error message for invalid email " Please enter a valid email address."

**Comment by klohman** [comment]
# Party Type: Guardian    

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "A minor or incompetent person", display the radio button options:
- Conservator 
- Guardian 
- Custodian 
- Next Friend for a Minor (Without a Guardian, Conservator, or other like Fiduciary)
- Next Friend for an Incompetent Person (Without a Guardian, Conservator, or other like Fiduciary)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/0c5bac07-c365-4a29-ac29-42d40d35aa23)

On select of "Guardian" radio button option 
- party type is set to "Guardian" 
- "Tell Us About Yourself as the Guardian for This Taxpayer" and " Tell Us About the Taxpayer You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/7e9fb585-a765-4805-a6de-243851aaa3e2)

### Tell Us About Yourself as the Guardian for This Taxpayer (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

## Tell Us About the Taxpayer You Are Filing For (Secondary Contact) 
### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 
- Email validation 
   - email must be valid email format (prefix@domain) 
   - Error message for invalid email " Please enter a valid email address."

**Comment by klohman** [comment]
# Party Type: Custodian   

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "A minor or incompetent person", display the radio button options:
- Conservator 
- Guardian 
- Custodian 
- Next Friend for a Minor (Without a Guardian, Conservator, or other like Fiduciary)
- Next Friend for an Incompetent Person (Without a Guardian, Conservator, or other like Fiduciary)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/cac0c20a-54a0-4b44-a15d-f94d537ade54)

On select of "Custodian" radio button option 
- party type is set to "Custodian" 
- "Tell Us About Yourself as the Custodian for This Taxpayer" and " Tell Us About the Taxpayer You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/a7a9b1e6-2be6-494b-8186-2a4b624b7eb7)

### Tell Us About Yourself as the Custodian for This Taxpayer (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

## Tell Us About the Taxpayer You Are Filing For (Secondary Contact) 
### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 
- Email validation 
   - email must be valid email format (prefix@domain) 
   - Error message for invalid email " Please enter a valid email address."

**Comment by klohman** [comment]
# Party Type: Minor (Without Guardian, Conservator, or other like Fiduciary)    

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "A minor or incompetent person", display the radio button options:
- Conservator 
- Guardian 
- Custodian 
- Next Friend for a Minor (Without a Guardian, Conservator, or other like Fiduciary)
- Next Friend for an Incompetent Person (Without a Guardian, Conservator, or other like Fiduciary)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/cac0c20a-54a0-4b44-a15d-f94d537ade54)

On select of "Minor (Without Guardian, Conservator, or other like Fiduciary)" radio button option 
- party type is set to "Minor (Without Guardian, Conservator, or other like Fiduciary)" 
- "Tell Us About Yourself as the Next Friend for This Minor" and " Tell Us About the Minor You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/bec63e03-c3b5-4c1f-8d99-5fbd35939908)

### Tell Us About Yourself as the Next Friend for This Minor (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

## Tell Us About the Minor You Are Filing For (Secondary Contact) 
### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 
- Email validation 
   - email must be valid email format (prefix@domain) 
   - Error message for invalid email " Please enter a valid email address."

**Comment by klohman** [comment]
# Party Type: Incompetent Person (Without Guardian, Conservator, or other like Fiduciary)

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "A minor or incompetent person", display the radio button options:
- Conservator 
- Guardian 
- Custodian 
- Next Friend for a Minor (Without a Guardian, Conservator, or other like Fiduciary)
- Next Friend for an Incompetent Person (Without a Guardian, Conservator, or other like Fiduciary)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/cac0c20a-54a0-4b44-a15d-f94d537ade54)

On select of "Incompetent Person (Without Guardian, Conservator, or other like Fiduciary)" radio button option 
- party type is set to "Incompetent Person (Without Guardian, Conservator, or other like Fiduciary)" 
- "Tell Us About Yourself as the Next Friend for This Incompetent Person" and " Tell Us About the Incompetent Person You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/d3ebeef7-4d20-4f5f-9b71-e33f78bed026)

### Tell Us About Yourself as the Next Friend for This Incompetent Person (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

## Tell Us About the Incompetent Person You Are Filing For (Secondary Contact) 
### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 
- Email validation 
   - email must be valid email format (prefix@domain) 
   - Error message for invalid email " Please enter a valid email address."

**Comment by klohman** [comment]
# Party Type: Donor

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "Donor" radio button option 
- party type is set to "Donor"
- "Tell Us About the Donor You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/f0c7f41d-907c-4077-a14f-1faa0df6e23a)

### Tell Us About the Donor You Are Filing For (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

**Comment by klohman** [comment]
# Party Type: Transferee 

On select of "Other", display radio button options: 
   - An estate or trust 
   - A minor or incompetent person 
   - Donor 
   - Transferee 
   - Deceased Spouse 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b484631c-b213-453b-acfb-b1473035dee5)

On select of "Transferee" radio button option 
- party type is set to "Transferee"
- "Tell Us About the Transferee You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/b3c1c158-b573-4cea-9da6-98c173d3409e)

### Tell Us About the Transferee You Are Filing For (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#"
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

**Comment by klohman** [comment]
# Party Type: Corporation  

On select of "A business", display radio button options: 
   - Corporation 
   - Partnership (as the tax matters partner)
   - Partnership (as a partner other than tax matters partner) 
   - Partnership (as a partnership representative under the BBA regime)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/e90d6c58-a5af-454f-a089-0f46cac9c101)

On select of "Corporation" radio button option 
- party type is set to "Corporation"
- "Tell Us About the Corporation You Are Filing For" views display 
   
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/5514d1e7-1cb2-4072-8bce-9d45d6b2d15a)

### Tell Us About the Corporation You Are Filing For (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

**Comment by klohman** [comment]
# Party Type: Partnership (Tax Matters Partner)  

On select of "A business", display radio button options: 
   - Corporation 
   - Partnership (as the tax matters partner)
   - Partnership (as a partner other than tax matters partner) 
   - Partnership (as a partnership representative under the BBA regime)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/e90d6c58-a5af-454f-a089-0f46cac9c101)

On select of "Partnership (as the tax matters partner" radio button option 
- party type is set to "Partnership (Tax Matters Partner)"
- " Tell Us About Yourself as the Tax Matters Partner" and "Tell Us About the Corporation You Are Filing For" views display 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/32c1ee62-e68e-4a7e-bac3-ae3c4995b724)

### Tell Us About Yourself as the Tax Matters Partner (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

### Tell Us About the Partnership You Are Filing For (Secondary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email must be in valid email format (prefix @domain)

**Comment by klohman** [comment]
# Party Type: Partnership (as a partner other than tax matters partner)  

On select of "A business", display radio button options: 
   - Corporation 
   - Partnership (as the tax matters partner)
   - Partnership (as a partner other than tax matters partner) 
   - Partnership (as a partnership representative under the BBA regime)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/e90d6c58-a5af-454f-a089-0f46cac9c101)

On select of "Partnership (partner other than tax matters partner" radio button option 
- party type is set to "Partnership (Partner Other Than Tax Matters Partner)"
- " Tell Us About Yourself as the Partner" and "Tell Us About the Partnership You Are Filing For" views display 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/a78b00a1-fdd9-4640-83a9-673acfb50196)

### Tell Us About Yourself as the Partner (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

### Tell Us About the Partnership You Are Filing For (Secondary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email must be in valid email format (prefix @domain)

**Comment by klohman** [comment]
# Party Type: Partnership (as a partnership representative under the BBA regime)

On select of "A business", display radio button options: 
   - Corporation 
   - Partnership (as the tax matters partner)
   - Partnership (as a partner other than tax matters partner) 
   - Partnership (as a partnership representative under the BBA regime)

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/e90d6c58-a5af-454f-a089-0f46cac9c101)

On select of "Partnership (partnership representative under the BBA regime" radio button option 
- party type is set to "Partnership (BBA Regime))"
- " Tell Us About Yourself as the Partner" and "Tell Us About the Partnership You Are Filing For" views display 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/53a0c696-e53a-4c93-9003-1c10e1178728)

### Tell Us About Yourself as the Partner (Primary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email will be pre-populated with the registered user's email
    - email will be read-only

### Tell Us About the Partnership You Are Filing For (Secondary Contact) 

### Field Validation 
-  All fields are required EXCEPT "Suite/Apt#" and "In Care Of" 
- Zip Code validation
   - must be either 5 digit or 5 AND 4
   - Numeric values and special characters (-) only 
  - Error message for invalid zip code "Please enter a valid zip code." 

- Email Validation
    - email must be in valid email format (prefix @domain)

**Comment by klohman** [comment]
# Case Detail Page - Case Information Tab 

Party Type information will display on the case detail page under the Case Information tab, using the following template 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/59904e68-05d3-41cc-b51c-4b6f8864fc63)

**Comment by klohman** [comment]
# Error Validation

on Click of Submit to U.S. Tax Court button - should validate
- all fields required  have been completed
- data is in the correct format for the following fields
   - IRS Notice Date 
   - Zip Code 
   - Email Address 

### Error messaging 

-  If any errors are found, display page level error message (as shown in example design) with list of all fields with errors 

- fields with errors should display in their field level error state with the following messaging: 
   - Empty required fields " [Field name] is required" 
   - Invalid zip code : "Please enter a valid zip code."
   - Invalid email address " "Please enter a valid email address"
   - Invalid IRS Notice date : "Please enter a valid date"
   - IRS Notice Date in future - "That date's in the future. Please enter a valid date"

Page level errors (both the warning at the top and the indicators by the field) remain until the form has been submitted again. If errors remain or new ones are found, page level errors load again. If form is error-free, form is submitted successfully.

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/9468c89e-e5bd-4a98-93da-2ca1fc7c6292)

**Comment by uxforgood** [comment]
@ericsorenson @rparris-flexion 

Here are our recommendations for mobile display of the hints within the start a case form. If any are difficult to accomplish, please let us know! 

1. Upload hint - display under "choose file" component

2. Filing on behalf of someone else hint - display under the "Who is Filing This Case?" header, before the radio button label and selections

3. Change email hint - display under email text, before the Phone Number field label

![image.png](https://images.zenhubusercontent.com/5bc8eb638ff5515ac40d1d90/959bbf4b-f1b6-4533-9181-10cb84d65dd7)

**Comment by ericsorenson** [comment]
Fixed to be applied to dev after the demo:

- [ ] Extra space between City and Zip
- [ ] Estate with an executor helper text should be below label
- [ ] Estate or trust contact, transferee, donor validators are not firing
- [ ] Conservator, Guardian, Custodian validation is incomplete 
- [ ] Run validation for selects, radios, checkboxes on change instead of on blur
- [ ] Show trial city error only after case procedure is elected
- [ ] Update language as described by Jessica
